server/routes/api/teas.js

```diff
@@ -33,65 +34,63 @@ router.post("/new-tea", (req, res) => {
 
     newTea
       .save()
-      .then(tea => res.json(tea))
+      .then(tea => res.json(teaNormalizer(tea)))
       .catch(err => console.log(err));
-
   });
 });
 
-
 router.put("/update-tea", (req, res) => {
-
-  Tea.findOneAndUpdate({id: req.body.id}, req.body, {new: true}, function(err, doc){
+  Tea.findOneAndUpdate({ id: req.body.id }, req.body, { new: true }, function(
+    err,
+    tea
+  ) {
     if (err) return res.send(500, { error: err });
-    return res.json(doc);
+    return res.json(teaNormalizer(tea));
   });
+});
 
-})
-
-router.delete("/delete-tea", (req, res) => {
+router.delete("/delete-tea/:id", (req, res) => {
   // The "tea" in this callback function represents the document that was found.
   // It allows you to pass a reference back to the client in case they need a reference for some reason.
-  Tea.findOneAndDelete({id: req.body.id},{}, (err, todo) => {
+  Tea.findOneAndDelete({ id: req.params.id }, {}, (err, todo) => {
     // As always, handle any potential errors:
     if (err) return res.status(500).send(err);
     // We'll create a simple object to send back with a message and the id of the document that was removed
     // You can really do this however you want, though.
     const response = {
-        message: "Tea successfully deleted",
-        id: req.body.id
+      message: "Tea successfully deleted",
+      id: req.body.id
     };
     return res.status(200).send(response);
   });
 });
 
 // get the tea with that id (accessed at GET http://localhost:8080/api/teas/:tea_id)
-router.get('/tea', (req, res) => {
-  Tea.findOne({id: req.body.id},(err, tea) => {
-    if (err){
+router.get("/tea", (req, res) => {
+  Tea.findOne({ id: req.body.id }, (err, tea) => {
+    if (err) {
       res.send(err);
     }
-    if(tea){
+    if (tea) {
       res.json(tea);
-    }else{
-      return res.json({ message: 'Tea does not exist.' });
+    } else {
+      return res.json({ message: "Tea does not exist." });
     }
   });
 });
 
+router.get("/teasList/:id", function(req, res) {
+  Tea.find({ userID: req.params.id }, function(err, teas) {
+    let teaMap = [];
+    let index = 0;
 
-
-router.get('/teasList/:id', function(req, res) {
-  Tea.find({userID: req.params.id}, function(err, teas) {
-    var teaMap = [];
-    var index = 0
     teas.forEach(function(tea) {
-      teaMap[index] = tea;
+      teaMap[index] = teaNormalizer(tea);
       index++;
     });
```

Comment:
This can be changed to `.map()` instead of `.forEach()`. Regardless, the index variable doesn't seem to be used.

Reply:
No pushing required. [`map` returns a new array with mapped values](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map) which means you don't need to manually create an array and push objects into it.

Reply:
would this be the correct format for this?
`let teaMap = teas.map(tea => {
      tea = teaNormalizer(tea)
    });`